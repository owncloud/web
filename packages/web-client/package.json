{
  "name": "@ownclouders/web-client",
  "version": "11.3.1",
  "description": "ownCloud web client",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-client",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-client"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "default": "./dist/web-client.js",
        "require": "./dist/web-client.cjs",
        "types": "./dist/src/index.d.ts"
      },
      "./graph": {
        "default": "./dist/web-client/graph.js",
        "require": "./dist/web-client/graph.cjs",
        "types": "./dist/src/graph/index.d.ts"
      },
      "./graph/generated": {
        "default": "./dist/web-client/graph/generated.js",
        "require": "./dist/web-client/graph/generated.cjs",
        "types": "./dist/src/graph/generated/index.d.ts"
      },
      "./ocs": {
        "default": "./dist/web-client/ocs.js",
        "require": "./dist/web-client/ocs.cjs",
        "types": "./dist/src/ocs/index.d.ts"
      },
      "./sse": {
        "default": "./dist/web-client/sse.js",
        "require": "./dist/web-client/sse.cjs",
        "types": "./dist/src/sse/index.d.ts"
      },
      "./webdav": {
        "default": "./dist/web-client/webdav.js",
        "require": "./dist/web-client/webdav.cjs",
        "types": "./dist/src/webdav/index.d.ts"
      }
    }
  },
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./graph": {
      "default": "./src/graph/index.ts"
    },
    "./graph/generated": {
      "default": "./src/graph/generated/index.ts"
    },
    "./ocs": {
      "default": "./src/ocs/index.ts"
    },
    "./sse": {
      "default": "./src/sse/index.ts"
    },
    "./webdav": {
      "default": "./src/webdav/index.ts"
    },
    "./l10n": {
      "default": "./l10n/translations.json"
    }
  },
  "scripts": {
    "generate-openapi": "rm -rf src/graph/generated && docker run --rm -v \"${PWD}/src/graph:/local\" openapitools/openapi-generator-cli generate -i https://raw.githubusercontent.com/owncloud/libre-graph-api/main/api/openapi-spec/v1.0.yaml  -g typescript-axios -o /local/generated",
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && rm -rf package/dist/tests && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@microsoft/fetch-event-source": "^2.0.1",
    "@ownclouders/web-pkg": "workspace:*",
    "axios": "^1.7.7",
    "fast-xml-parser": "^4.5.0",
    "lodash-es": "^4.17.21",
    "luxon": "^3.5.0",
    "uuid": "^11.0.0",
    "webdav": "^5.7.1",
    "xml-js": "^1.6.11",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@types/luxon": "3.4.2",
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "5.0.0",
    "vite-plugin-dts": "4.2.4",
    "vite-plugin-node-polyfills": "0.22.0",
    "vite": "5.4.8"
  }
}
