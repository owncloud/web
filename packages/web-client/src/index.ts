import { graph } from './graph'
import { ocs } from './ocs'
import { webdav } from './webdav'

export * from './errors'
export * from './helpers'
export * from './utils'
export * from './constants'

export { graph, ocs, webdav }
