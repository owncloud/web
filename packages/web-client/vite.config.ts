import { join, resolve } from 'path'
import { defineConfig, searchForWorkspaceRoot } from 'vite'
import dts from 'vite-plugin-dts'
import { nodePolyfills } from 'vite-plugin-node-polyfills'

const projectRootDir = searchForWorkspaceRoot(process.cwd())

export default defineConfig({
  resolve: {
    alias: {
      crypto: join(projectRootDir, 'polyfills/crypto.js')
    }
  },
  build: {
    lib: {
      entry: {
        'web-client': resolve(__dirname, 'src/index.ts'),
        'web-client/graph': resolve(__dirname, 'src/graph/index.ts'),
        'web-client/graph/generated': resolve(__dirname, 'src/graph/generated/index.ts'),
        'web-client/ocs': resolve(__dirname, 'src/ocs/index.ts'),
        'web-client/sse': resolve(__dirname, 'src/sse/index.ts'),
        'web-client/webdav': resolve(__dirname, 'src/webdav/index.ts')
      }
    }
  },
  plugins: [
    nodePolyfills({
      exclude: ['crypto']
    }),
    dts({
      include: ['src'],
      outDir: 'dist/types',
      insertTypesEntry: true
    })
  ]
})
