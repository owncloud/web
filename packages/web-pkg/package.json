{
  "name": "@ownclouders/web-pkg",
  "version": "11.1.3",
  "description": "ownCloud web pkg",
  "license": "AGPL-3.0",
  "main": "src/index.ts",
  "types": "types.d.ts",
  "private": false,
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "type": "module",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-pkg",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-pkg"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "import": "./dist/web-pkg.js",
        "require": "./dist/web-pkg.umd.cjs",
        "types": "./dist/src/index.d.ts"
      }
    }
  },
  "scripts": {
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && rm -rf package/dist/tests package/dist/*.css && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@casl/vue": "^2.2.2",
    "@microsoft/fetch-event-source": "^2.0.1",
    "@ownclouders/design-system": "workspace:^",
    "@ownclouders/web-client": "workspace:^",
    "@sentry/vue": "^8.33.1",
    "@uppy/core": "^3.3.0",
    "@uppy/drop-target": "^2.0.0",
    "@uppy/tus": "^3.1.0",
    "@uppy/utils": "^5.3.0",
    "@uppy/xhr-upload": "^3.0.1",
    "@vavt/cm-extension": "^1.6.0",
    "@vue/shared": "^3.5.11",
    "@vueuse/core": "^11.0.0",
    "axios": "^1.7.7",
    "deepmerge": "^4.2.2",
    "dompurify": "^3.2.3",
    "filesize": "^10.1.0",
    "fuse.js": "^7.0.0",
    "js-generate-password": "^1.0.0",
    "lodash-es": "^4.17.21",
    "luxon": "^3.5.0",
    "mark.js": "^8.11.1",
    "md-editor-v3": "^5.2.1",
    "oidc-client-ts": "^2.4.0 || ^3.0.0",
    "p-queue": "^8.0.0",
    "password-sheriff": "^1.1.1",
    "pinia": "^2.2.4",
    "portal-vue": "^3.0.0",
    "prismjs": "^1.29.0",
    "qs": "^6.13.0",
    "semver": "^7.6.3",
    "uuid": "^11.0.0",
    "vue-concurrency": "^5.0.1",
    "vue-router": "^4.2.5",
    "vue3-gettext": "^2.4.0",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@ownclouders/web-test-helpers": "workspace:^",
    "@types/dompurify": "3.0.5",
    "@types/lodash-es": "4.17.12",
    "@vitest/web-worker": "2.1.2",
    "clean-publish": "5.0.0",
    "vite-plugin-dts": "4.2.4",
    "vite-plugin-node-polyfills": "0.22.0"
  }
}
