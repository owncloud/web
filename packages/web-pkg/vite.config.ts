import { join, resolve } from 'path'
import { defineConfig, searchForWorkspaceRoot } from 'vite'
import dts from 'vite-plugin-dts'
import { nodePolyfills } from 'vite-plugin-node-polyfills'
import vue from '@vitejs/plugin-vue'
import pkg from './package.json' assert { type: 'json' }

const projectRootDir = searchForWorkspaceRoot(process.cwd())
const external = [...Object.keys(pkg.dependencies)]

export default defineConfig({
  resolve: {
    alias: {
      crypto: join(projectRootDir, 'polyfills/crypto.js')
    }
  },
  css: {
    preprocessorOptions: {
      scss: {
        additionalData: `
          @use "sass:math";
          @import "${projectRootDir}/packages/design-system/src/styles/styles";
        `,
        silenceDeprecations: ['legacy-js-api', 'import']
      }
    }
  },
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'web-pkg',
      fileName: 'web-pkg'
    },
    rollupOptions: {
      external
    }
  },
  plugins: [
    vue(),
    nodePolyfills({
      exclude: ['crypto']
    }),
    dts({ exclude: ['**/tests'], include: ['src'], outDir: 'dist/types', insertTypesEntry: true }),
    {
      name: '@ownclouders/vite-plugin-docs',
      transform(src, id) {
        if (id.includes('type=docs')) {
          return {
            code: 'export default {}',
            map: null
          }
        }
      }
    }
  ]
})
