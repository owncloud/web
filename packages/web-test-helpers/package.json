{
  "name": "@ownclouders/web-test-helpers",
  "version": "11.3.2",
  "description": "ownCloud web test helpers",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "main": "src/index.ts",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-test-helpers",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-test-helpers"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "import": "./dist/web-test-helpers.es.js",
        "require": "./dist/web-test-helpers.umd.cjs",
        "types": "./dist/src/index.d.ts"
      }
    }
  },
  "scripts": {
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "peerDependencies": {
    "@ownclouders/web-pkg": "workspace:^",
    "@vue/test-utils": "^2.4.6",
    "vue": "^3.5.10"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "^5.0.0",
    "vite-plugin-dts": "^4.2.3",
    "vite": "^5.4.8"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@casl/vue": "^2.2.2",
    "@ownclouders/design-system": "workspace:^",
    "@ownclouders/web-client": "workspace:^",
    "@pinia/testing": "^0.1.3",
    "axios": "1.7.7",
    "pinia": "2.2.6",
    "vitest-mock-extended": "2.0.2",
    "vue-router": "4.2.5",
    "vue3-gettext": "2.4.0"
  }
}
