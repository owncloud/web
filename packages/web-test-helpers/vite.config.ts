import { resolve } from 'path'
import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'
import pkg from './package.json' assert { type: 'json' }
import vue from '@vitejs/plugin-vue'

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'web-test-helpers',
      fileName: (format) => `web-test-helpers.${format}.js`
    },
    rollupOptions: {
      external: [...Object.keys(pkg.dependencies), ...Object.keys(pkg.peerDependencies)]
    }
  },
  plugins: [vue(), dts({ include: ['src'], outDir: 'dist/types', insertTypesEntry: true })]
})
