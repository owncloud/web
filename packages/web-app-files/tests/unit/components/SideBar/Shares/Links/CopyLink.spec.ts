import { ref } from 'vue'
import { LinkShare } from '@ownclouders/web-client'
import { useMessages } from '@ownclouders/web-pkg'
import CopyLink from '../../../../../../src/components/SideBar/Shares/Links/CopyLink.vue'
import { useClipboard } from '@vueuse/core'
import { mock } from 'vitest-mock-extended'
import { defaultPlugins, mount } from '@ownclouders/web-test-helpers'

const linkShare = {
  displayName: 'Example link',
  webUrl: 'https://some-url.com/abc'
} as LinkShare

vi.mock('@vueuse/core', () => ({
  useClipboard: vi.fn(() => ({
    copy: vi.fn(),
    copied: false,
    isSupported: true
  }))
}))

describe('CopyLink', () => {
  // ignore tippy warning
  vi.spyOn(console, 'warn').mockImplementation(undefined)
  it('upon clicking it should copy the link to the clipboard, render a success message and change icon for half a second', async () => {
    const copyMock = vi.fn()
    const copiedRef = ref(true)
    vi.mocked(useClipboard).mockReturnValue(
      mock<ReturnType<typeof useClipboard>>({ copy: copyMock, copied: copiedRef })
    )

    const { wrapper } = getWrapper()
    const { showMessage } = useMessages()
    expect(showMessage).not.toHaveBeenCalled()

    await wrapper.find('.oc-files-public-link-copy-url').trigger('click')
    expect(copyMock).toHaveBeenCalledTimes(1)
    expect(wrapper.html()).toMatchSnapshot()
    expect(showMessage).toHaveBeenCalledTimes(1)

    copiedRef.value = false

    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
})

function getWrapper() {
  return {
    wrapper: mount(CopyLink, {
      props: {
        linkShare
      },
      global: {
        plugins: [...defaultPlugins()],
        stubs: {
          'oc-icon': {
            template: '<span class="oc-icon" />'
          }
        }
      }
    })
  }
}
