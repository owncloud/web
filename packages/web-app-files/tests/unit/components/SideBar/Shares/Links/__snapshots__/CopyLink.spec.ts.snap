// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CopyLink > upon clicking it should copy the link to the clipboard, render a success message and change icon for half a second 1`] = `
"<button type="button" aria-label="Copy link to clipboard" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-public-link-copy-url">
  <!--v-if-->
  <!-- @slot Content of the button --> <span class="oc-icon" name="checkbox-circle" fill-type="line"></span>
</button>"
`;

exports[`CopyLink > upon clicking it should copy the link to the clipboard, render a success message and change icon for half a second 2`] = `
"<button type="button" aria-label="Copy link to clipboard" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-public-link-copy-url">
  <!--v-if-->
  <!-- @slot Content of the button --> <span class="oc-icon" name="file-copy" fill-type="line"></span>
</button>"
`;
